language/move-vm/types/src/values/values_impl.rs

```diff
@@ -64,7 +64,7 @@ enum ValueImpl {
 enum Container {
     Locals(Rc<RefCell<Vec<ValueImpl>>>),
     Vec(Rc<RefCell<Vec<ValueImpl>>>),
-    Struct(Rc<RefCell<Vec<ValueImpl>>>),
+    Struct(Rc<RefCell<Vec<ValueImpl>>>, Option<u64>),
```

Comment:
Again, I think we should move this into the `Rc` or otherwise `Value` is gonna become 16B larger

Reply:
This could be a much bigger change cuz I think the vm value is using `Rc<RefCell<Vec<ValueImpl>>>` everywhere.. changing this could yield a bigger refactor.

Reply:
> changing this could yield a bigger refactor

You will need to borrow the refcell in order to access the tag, but I don't think it will result in architectural changes. I'm concerned that if we end up adding more info/tags in the future, then it's gonna have a devastating effect on the value stack in terms of performance.